ext/src/string_extension.cr

```diff
@@ -0,0 +1,76 @@
+# exact port from ActiveSupport in Ruby:
+# https://github.com/rails/rails/blob/42b0c942520e59399d70c2170253aa5275a42af1/activesupport/lib/active_support/core_ext/object/blank.rb#L101-L119
+class String
+  BLANK_RE = /\A[[:space:]]*\z/
+
+  # A string is blank if it's empty or contains whitespaces only:
+  #
+  #   ''.blank?       # => true
+  #   '   '.blank?    # => true
+  #   "\t\n\r".blank? # => true
+  #   ' blah '.blank? # => false
+  #
+  # Unicode whitespace is supported:
+  #
+  #   "\u00a0".blank? # => true
+  #
+  # @return [true, false]
+  def slow_blank?
+    BLANK_RE === self
+  end
+
+  def blank_as?
+    return true if self.nil? || self.size == 0
+    i = 0
+    while i < self.size
+      case self[i]
+      when "\u0009",
```

Comment:
@akitaonrails This is a string literal, it will never match a character. Maybe you meant to use single quotes?

As a side note, iterating this way is slow. Use `self.each_char`. On the other hand, Ruby strings are not always UTF-8 encoded, while Crystal strings are, so this will fail for non-UTF-8 Ruby strings. 

The way to correctly do this is by implementing something like the C code, using Ruby function calls, pointers and integers.